packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -57,20 +57,15 @@ export const isValidCodeWhispererCoreConnection = (conn?: Connection): conn is C
         (isSsoConnection(conn) && hasScopes(conn, codeWhispererCoreScopes))
     )
 }
-/** For Builder ID only, if using IdC then use {@link isValidAmazonQConnection} */
-export const isValidCodeWhispererChatConnection = (conn?: Connection): conn is Connection => {
+/** Superset that includes all of CodeWhisperer + Amazon Q */
+export const isValidAmazonQConnection = (conn?: Connection): conn is Connection => {
     return (
-        isBuilderIdConnection(conn) &&
+        (isSsoConnection(conn) || isBuilderIdConnection(conn)) &&
         isValidCodeWhispererCoreConnection(conn) &&
-        hasScopes(conn, codeWhispererChatScopes)
```

Comment:
Ensure we are not using `codeWhispererChatScopes` or `codeWhispererCoreScopes` outside of using it to create `amazonQScopes`. Only `amazonQScopes` should have uses in the code.